joi-validations/chat.validations.js

```diff
@@ -1,6 +1,13 @@
 const {uuid, array, object, string} = require('./general.validations');
 
 const ChatValidation = {
+  changeChannelDetail: {
+    body: {
+      channel_id: string.required(),
+      channel_name: string.optional(),
+      channel_image: string.optional()
+    }
+  },
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The validation schema for `changeChannelDetail` seems to be missing a check for the presence of at least one of `channel_name` or `channel_image`. As it stands, a request could pass validation without either field, which would make the update operation meaningless. Consider adding a custom validation rule to ensure that at least one of these fields is present in the request.

```diff
  changeChannelDetail: {
    body: {
      channel_id: string.required(),
      channel_name: string.optional(),
      channel_image: string.optional()
+     .custom((value, helpers) => {
+       if (!value.channel_name && !value.channel_image) {
+         return helpers.message('At least one of channel_name or channel_image must be provided');
+       }
+       return value;
+     })
    }
  },
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->